### PR TITLE
grafana/ui : Introduce WithSeriesOptions interface (alt to #16565)

### PR DESCRIPTION
**Pull request by dprokop**
Alternative to #16565

### CODE REVIEW

File: public/app/features/dashboard/state/PanelModel.ts [hunk 1]
```diff
@@ -130,6 +131,7 @@ export class PanelModel {
     this.ensureQueryIds();
 
     this.restoreInfintyForThresholds();
+    this.updateOptions = this.updateOptions.bind(this);
```
Comment (torkelo) [inline public/app/features/dashboard/state/PanelModel.ts]: bit sure I like this bind, due use from outside. this should be handled in PanelChrome instead it  is the component that sets the onOptionsChange

File: public/app/features/dashboard/panel_editor/QueriesTab.tsx [hunk 2]
```diff
@@ -113,6 +113,10 @@ export class QueriesTab extends PureComponent<Props, State> {
 
     const index = _.indexOf(panel.targets, query);
     panel.targets.splice(index, 1);
+
+    if ('series' in panel.options) {
```
Comment (torkelo) [inline public/app/features/dashboard/panel_editor/QueriesTab.tsx]: think we can remove this.

File: packages/grafana-ui/src/types/panel.ts [hunk 3]
```diff
@@ -5,11 +5,16 @@ import { ScopedVars } from './datasource';
 
 export type InterpolateFunction = (value: string, scopedVars?: ScopedVars, format?: string | Function) => string;
 
+export type WithSeriesOptions<T extends {}, S> = T & {
```
Comment (torkelo) [inline packages/grafana-ui/src/types/panel.ts]: I am not sure this interface will work, not every options on a panel will have a series override.

**Review by ryantxu — CHANGES_REQUESTED** [review]
Oops - I left the review pending

**Comment by dprokop** [comment]
@ryantxu closing this one and #16565 as well for now. We talked with @torkelo and none of these solutions are satisfying for now, so we gonna explore this subject a little bit more.

**Pull request closed without merging**